pkg/apis/alibabacloudprovider/v1beta1/alibabacloudmachineproviderconfig_types.go

```diff
@@ -247,6 +179,177 @@ type AlibabaCloudMachineProviderConfigList struct {
+	Name string `name:"name,omitempty"`
+
+	//The ID of the snapshot used to create data disk N. Valid values of N: 1 to 16.
+	//
+	//When the DataDisk.N.SnapshotID parameter is specified, the DataDisk.N.Size parameter is ignored. The data disk is created based on the size of the specified snapshot.
+	//Use snapshots created after July 15, 2013. Otherwise, an error is returned and your request is rejected.
+	// +optional
+	SnapshotID string `name:"snapshotId,omitempty"`
+
+	//The size of data disk N. Valid values of N: 1 to 16. Unit: GiB. Valid values:
+	//
+	//Valid values when DataDisk.N.Category is set to cloud_efficiency: 20 to 32768
+	//Valid values when DataDisk.N.Category is set to cloud_ssd: 20 to 32768
+	//Valid values when DataDisk.N.Category is set to cloud_essd: 20 to 32768
+	//Valid values when DataDisk.N.Category is set to cloud: 5 to 2000
+	//The value of this parameter must be greater than or equal to the size of the snapshot specified by the SnapshotID parameter.
+	// +optional
+	Size int `name:"size,omitempty"`
+
+	//Specifies whether to encrypt data disk N.
+	//
+	//Default value: false.
+	// +optional
+	Encrypted bool `name:"encrypted,omitempty"`
+
+	//
+	//The performance level of the ESSD used as data disk N. The N value must be the same as that in DataDisk.N.Category when DataDisk.N.Category is set to cloud_essd. Default value: PL1. Valid values:
+	//
+	//PL0: A single ESSD can deliver up to 10,000 random read/write IOPS.
+	//PL1: A single ESSD can deliver up to 50,000 random read/write IOPS.
+	//PL2: A single ESSD can deliver up to 100,000 random read/write IOPS.
+	//PL3: A single ESSD can deliver up to 1,000,000 random read/write IOPS.
+	//For more information about ESSD performance levels, see ESSDs.
+	// +optional
+	PerformanceLevel string `name:"performanceLevel,omitempty"`
+
+	//TODO
+	//EncryptAlgorithm string `name:"EncryptAlgorithm"`
+
+	//The description of data disk N. The description must be 2 to 256 characters in length and cannot start with http:// or https://.
+	//
+	//This parameter is empty by default.
+	// +optional
+	Description string `name:"description,omitempty"`
+
+	//The category of data disk N. Valid values:
+	//
+	//cloud_efficiency: ultra disk
+	//cloud_ssd: standard SSD
+	//cloud_essd: ESSD
+	//cloud: basic disk
+	//For I/O optimized instances, the default value is cloud_efficiency. For non-I/O optimized instances, the default value is cloud.
+	// +optional
+	Category string `name:"category,omitempty"`
+
+	//The ID of the Key Management Service (KMS) key to be used by data disk N.
+	// +optional
+	KMSKeyID string `name:"kmsKeyId,omitempty"`
+
+	//The mount point of data disk N.
+	// +optional
+	Device string `name:"device,omitempty"`
+
+	//Specifies whether to release data disk N along with the instance.
+	//
+	//Default value: true.
+	// +optional
+	DeleteWithInstance *bool `name:"deleteWithInstance,omitempty"`
+}
+
+// Subscription information for the instance
+type SubscriptionInfo struct {
```

Comment:
Out of interest, not to do with the API, but what happens if the subscription for an instance runs out? Will Alibaba terminate the instance?

Reply:
@menglingwei Could you provide some insight on this? What happens when subscriptions expire?

Reply:
@kwoodson when subscriptions are expired. The ecs instance will be stopped. If you do not renew , the instance will be released after some days(15days or more ,i need to double check it)

Reply:
@kwoodson  For more informations about subscriptions ,you can link to [help documentions](https://www.alibabacloud.com/help/doc-detail/56220.htm?spm=a2c63.p38356.879954.5.5ad77a56lYhtIg#subs-china)

Reply:
![image](https://user-images.githubusercontent.com/2653769/140609211-323e287d-0930-4fda-85d7-23184fb7f0c4.png)

Reply:
@kwoodson  In OCP,which billing method will provide for end users ?

Reply:
@menglingwei I'm not sure which method. What do you recommend?

Reply:
@kwoodson  I suggest pay-as-you-go.